vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+    "at most one NE in between, and no iPR after iCR"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  order = exprs(ADT),
+  first_cond = AVALC.join %in% c("iCR", "iPR") &
+    ADT.join >= ADT + days(28),
+  condition = AVALC == "iPR" &
+    all(AVALC.join %in% c("iCR", "iPR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1 &
+    (
+      min_cond(
+        var = ADT.join,
+        cond = AVALC.join == "iCR"
+      ) > max_cond(var = ADT.join, cond = AVALC.join == "iPR") |
+        count_vals(var = AVALC.join, val = "iCR") == 0 |
+        count_vals(var = AVALC.join, val = "iPR") == 0
+    ),
+  set_values_to = exprs(AVALC = "Y")
+)
+```
+[^3]:
+```{r}
+icbor_icr <- event_joined(
+  description = paste(
+    "Define complete response (iCR) for confirmed best overall response (ICBOR) as",
+    "iCR followed by iCR at least 28 days later and at most one NE in between"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond = AVALC.join == "iCR" &
+    ADT.join >= ADT + 28,
+  condition = AVALC == "iCR" &
+    all(AVALC.join %in% c("iCR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1,
+  set_values_to = exprs(AVALC = "iCR")
+)
+```
+[^4]:
+```{r}
+icbor_ipr <- event_joined(
+  description = paste(
+    "Define partial response (iPR) for confirmed best overall response (ICBOR) as",
+    "iPR followed by iCR or iPR at least 28 days later, at most one NE in between,",
+    "and no iPR after iCR"
+  ),
+  dataset_name = "ovr",
+  join_vars = exprs(AVALC, ADT),
+  join_type = "after",
+  first_cond = AVALC.join %in% c("iCR", "iPR") &
+    ADT.join >= ADT + 28,
+  condition = AVALC == "iPR" &
+    all(AVALC.join %in% c("iCR", "iPR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1 &
+    (
+      min_cond(
+        var = ADT.join,
+        cond = AVALC.join == "iCR"
+      ) > max_cond(var = ADT.join, cond = AVALC.join == "iPR") |
+        count_vals(var = AVALC.join, val = "iCR") == 0 |
+        count_vals(var = AVALC.join, val = "iPR") == 0
+    ),
+  set_values_to = exprs(AVALC = "iPR")
+)
+```
+
+Please note that the result `AVALC = "Y"` for confirmed clinical benefit is
+defined by the first _two_ events specified for `events`. For subjects with
+observations fulfilling both events the one with the earlier date should be
+selected (and not the first one in the list). Thus `ignore_event_order = TRUE`
+is specified.
+
+```{r}
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(icrsp_y_icr, icrsp_y_ipr, no_data_n),
+    ignore_event_order = TRUE,
+    set_values_to = exprs(
+      PARAMCD = "ICRSP",
+      PARAM = "Immune Response by Investigator (confirmation required)",
```

Comment:
I would keep consistency with adrs.html and would assign PARAM = "Confirmed Response by Investigator"

Reply:
will update